src/commands/deploy/utils.ts

```diff
@@ -125,6 +137,168 @@ export async function getOrCreateDatabase(
     };
 }
 
+export async function enableAuthentication(
+    configuration: YamlProjectConfiguration,
+    projectId: string,
+    projectEnvId: string,
+    stage: string,
+    envFile: string | undefined,
+) {
+    const authDatabase = configuration.services?.authentication?.database;
+    if (!authDatabase) {
+        return;
+    }
+
+    const authProviders = configuration.services?.authentication
+        ?.providers as AuthenticationProviders;
+
+    if ("uri" in authDatabase && "type" in authDatabase) {
+        const databaseUri = await evaluateResource(configuration, authDatabase.uri, stage, envFile);
+
+        await enableAuthenticationHelper(
+            {
+                enabled: true,
+                databaseUri: databaseUri,
+                databaseType: authDatabase.type,
+            },
+            projectEnvId,
+            authProviders,
+        );
+    }
+    if ("name" in authDatabase && "region" in authDatabase && "type" in authDatabase) {
+        const database: GetDatabaseResponse = await getOrCreateDatabase(
+            {
+                name: authDatabase.name,
+                region: authDatabase.region,
+                type: authDatabase.type,
+            },
+            stage,
+            projectId,
+            projectEnvId,
+        );
+
+        await enableAuthenticationHelper(
+            {
+                enabled: true,
+                databaseUri: database.connectionUrl || "",
+                databaseType: database.type,
+            },
+            projectEnvId,
+            authProviders,
+        );
+    }
```

Comment:
Maybe put an else here that covers the bad path snd throws an error. Otherwise, if it does not enter neither ifs, this code will silently fail